### PR TITLE
Polio 614 budget tab update

### PR DESCRIPTION
**Pull request by quang-le**
Needed to add fields from the budget tool to campaign's budget tab 

## Self proof reading checklist

- [X] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [ ] Are my typescript files well typed
- [X] Did I add translations
- [X] My migrations file are included
- [ ] Are there enough tests
## Changes

- Added the needed field `Campaign` model.
- Updated status dropdown list in in budget tab to use states from [current workflow](https://drive.google.com/drive/u/0/folders/1ua4yzyFOGayweQVNKP0hyBNDKhQrPlx1) 
- Added `ExpandableItem` component to make long list of inputs expandable
- Added the fields in the budget tab
- Reorganised budget tab layout to separate budget approval, funds release and cost per child
- Added new fields to yup validation

## How to test

Go to polio campaigns and play with budget tab

## Print screen / video

Updated dropdown
Screenshot 2022-12-21 at 18 36 06

Collapsable fields

https://user-images.githubusercontent.com/38907762/208969023-e7cd751d-d30c-433b-a5c2-731ef678556e.mov

## Notes

The funds release part of the tab has just been moved but otherwise hasn't been updated as we still expect feedback on some parts, and the budget tool hasn't been updated yet to handle funds release

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/app/components/ExpandableItem.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                </ListItemText>
+                {(open || preventCollapse) && <ExpandLess />}
+                {!open && !preventCollapse && <ExpandMore />}
+            </ListItem>
+            <Collapse in={open || preventCollapse} timeout="auto" unmountOnExit>
+                {children}
+            </Collapse>
+        </>
+    );
+};
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/domains/app/components/ExpandableItem.tsx:45]: Wan we create a ticket to put this on bluesquare-components please ?

File: hat/assets/js/apps/Iaso/utils/forms.js [hunk 2]
```diff
@@ -71,3 +73,8 @@ export const isFormValid = (requiredFields, currentForm) => {
             ),
     );
 };
+
+export const hasFormikFieldError = (key, errors, touched) => {
+    if (!errors) return false;
+    return Boolean(get(errors, key) && get(touched, key));
```
Comment (beygorghor) [inline hat/assets/js/apps/Iaso/utils/forms.js:79]: ❤️

File: plugins/polio/js/src/forms/BudgetForm.js [hunk 3]
```diff
@@ -47,26 +66,164 @@ export const BudgetForm = () => {
...
             <Grid container spacing={2}>
                 <Grid container direction="row" item spacing={2}>
                     <Grid xs={12} md={6} item>
+                        <Box mb={2}>
+                            <Typography variant="button">
+                                {formatMessage(MESSAGES.budgetApproval)}
+                            </Typography>
+                        </Box>
+                        <Box mb={2}>
+                            <Divider style={{ height: '1px', width: '100%' }} />
```
Comment (beygorghor) [inline plugins/polio/js/src/forms/BudgetForm.js:97]: is this style required ?

File: plugins/polio/js/src/forms/BudgetForm.js [hunk 4]
```diff
@@ -47,26 +66,164 @@ export const BudgetForm = () => {
...
                         />
-                    </Grid>
-                    <Grid xs={12} md={6} item>
+                    </Box>
+                    <Box mt={2}>
+                        <Divider style={{ height: '1px', width: '100%' }} />
+                    </Box>
+                    <ExpandableItem
+                        label={formatMessage(MESSAGES.budgetRequest)}
+                        preventCollapse={hasRequestFieldsError}
```
Comment (beygorghor) [inline plugins/polio/js/src/forms/BudgetForm.js:134]: 👍

File: plugins/polio/js/src/forms/BudgetForm.js [hunk 5]
```diff
@@ -47,26 +66,164 @@ export const BudgetForm = () => {
...
+                    >
+                        {BUDGET_REQUEST.map((node, index) => {
+                            return (
+                                <Box mt={index === 0 ? 2 : 0}>
```
Comment (beygorghor) [inline plugins/polio/js/src/forms/BudgetForm.js]: key is missing here !

File: plugins/polio/js/src/forms/BudgetForm.js [hunk 6]
```diff
@@ -47,26 +66,164 @@ export const BudgetForm = () => {
...
+                        })}
+                    </ExpandableItem>
+                    <Divider style={{ height: '1px', width: '100%' }} />
+                    <ExpandableItem
+                        label={formatMessage(MESSAGES.RRTReview)}
+                        preventCollapse={hasRRTReviewError}
+                    >
+                        {RRT_REVIEW.map((node, index) => {
+                            return (
+                                <Box mt={index === 0 ? 2 : 0}>
```
Comment (beygorghor) [inline plugins/polio/js/src/forms/BudgetForm.js]: same

File: plugins/polio/js/src/forms/BudgetForm.js [hunk 7]
```diff
@@ -47,26 +66,164 @@ export const BudgetForm = () => {
...
+                    <Box mt={2}>
+                        <Divider style={{ height: '1px', width: '100%' }} />
+                    </Box>
+                    <ExpandableItem
+                        label={formatMessage(MESSAGES.ORPGReview)}
+                        preventCollapse={hasORPGReviewError}
+                    >
+                        {ORPG_REVIEW.map((node, index) => {
+                            return (
+                                <Box mt={index === 0 ? 2 : 0}>
```
Comment (beygorghor) [inline plugins/polio/js/src/forms/BudgetForm.js]: key

File: plugins/polio/js/src/forms/BudgetForm.js [hunk 8]
```diff
@@ -47,26 +66,164 @@ export const BudgetForm = () => {
...
+                    <Box>
+                        <Divider style={{ height: '1px', width: '100%' }} />
+                    </Box>
+                    <ExpandableItem
+                        label={formatMessage(MESSAGES.approval)}
+                        preventCollapse={hasApprovalFieldsError}
+                    >
+                        {REVIEW_FOR_APPROVAL.map((node, index) => {
+                            return (
+                                <Box mt={index === 0 ? 2 : 0}>
```
Comment (beygorghor) [inline plugins/polio/js/src/forms/BudgetForm.js]: ...

**Review by beygorghor — CHANGES_REQUESTED** [review]
Better with expand / collapse !
See keys at the wrong places creating warnings.
Also i'm not sure this is the expected behaviour ? 
Screenshot 2022-12-22 at 12 53 01

File: plugins/polio/js/src/forms/BudgetForm.js (hunk 3 above)
Comment (quang-le) [reply plugins/polio/js/src/forms/BudgetForm.js:97]: I tried without, the `Divider`was invisible

File: plugins/polio/js/src/forms/BudgetForm.js (hunk 5 above)
Comment (quang-le) [reply plugins/polio/js/src/forms/BudgetForm.js]: Oops

File: plugins/polio/js/src/forms/BudgetForm.js (hunk 6 above)
Comment (quang-le) [reply plugins/polio/js/src/forms/BudgetForm.js]: I did it again!

File: plugins/polio/js/src/forms/BudgetForm.js (hunk 8 above)
Comment (quang-le) [reply plugins/polio/js/src/forms/BudgetForm.js]: ... and I've lost in that game

**Comment by quang-le** [comment]
> Better with expand / collapse ! See keys at the wrong places creating warnings. Also i'm not sure this is the expected behaviour ? Screenshot 2022-12-22 at 12 53 01

The error state with the dates is consistent with the behaviour throughout the polio form see rounds:


https://user-images.githubusercontent.com/38907762/209161481-05abb2e7-70e4-4b83-82cd-f568880d4cbb.mov

File: hat/assets/js/apps/Iaso/domains/app/components/ExpandableItem.tsx (hunk 1 above)
Comment (quang-le) [reply hat/assets/js/apps/Iaso/domains/app/components/ExpandableItem.tsx:45]: https://bluesquare.atlassian.net/browse/IA-1798

**Review by beygorghor — APPROVED** [review]

**Pull request merged**